integration/feature/bom/resources/build.mill

```diff
@@ -0,0 +1,200 @@
+package build
```

Comment:
I don't think this needs to be an integration test. Let's cover all the edge cases in a unit test, and have a single example test under `example/fundamentals/dependencies/` as documentation for the happy paths.

Reply:
Still working on the examples / document update.

That being said, I'm not sure we need such end-to-end, or at least, not in the documentation. BOM / dependency management only impacts dependency resolution, and the unit tests assert that we end up with the right JARs in `compileClasspath` and `runClasspath` (and in the generated `ivy.xml` and POM files too).

Also, it's likely the documentation pages for Scala and Kotlin will just refer to the Java one, as there's almost specific to Scala and Kotlin in BOM / dependency management.

Reply:
I would add one integration test for each language just in case

Reply:
I think having one example test per language makes sense here, but if you want to go with a single shared example test you can put it in `fundamentals/library-deps.adoc` which contains the "shared" dependency management documentation

Reply:
Just went for that